### ISSUE TITLE
LinkMap Key inconsistency between insert and select

### ISSUE THREAD

**Issue by guglielmomoretti** [orientechnologies/orientdb#5587]
The Linkmap type supports key range lookups by writing queries like

``` SQL
SELECT map["1-10"] from X
```

But what if the key itself is "1-10"? this force us to convert all minus sign in our map's key to underscore  to avoid this behaviour.  

If the key is "A-1" then the query will result in a NumberFormatException because "A" it's not a number

The following snippet rises the exception

``` SQL
CREATE CLASS Test;
CREATE CLASS TestMap;
CREATE PROPERTY Test.map LINKMAP TestMap;

INSERT INTO TestMap (name) VALUES ('ciao');
INSERT INTO Test (map) VALUES ({'A-1': <#RID of the previous record>});
SELECT map['A-1'] FROM Test;
```

the last line results in: **java.lang.NumberFormatException: For input string: "A"**

Would it be better if the parser could do **exact key match** if the key is between quotes  and perform **range queries** if the map key is not between quotes?

Ex

``` SQL
SELECT map["A-1"] from X
```

For an exact key match or:

``` SQL
SELECT map[1-3] from X
```

for a key range lookup?
(This could be linked also to #5588)

Regards

**Comment by luigidellaquila** [comment]
Hi @guglielmomoretti 

it seems to be a bug, I'll look into it

Thanks

Luigi

**Comment by busbaby** [comment]
Can someone provide a workaround for this? This one is biting me. For example; I'm unable to lookup java.util.UUID strings (ie: select map["8d2eb047-d372-45e7-b74f-050fda046b33"]).

**Comment by guglielmomoretti** [comment]
Remove the dashes or hash it. This is the only workaround I found...

Il 11 feb 2017 12:47, "busbaby" <notifications@github.com> ha scritto:

**Comment by busbaby** [comment]
Thanks for the quick reply. Unfortunately I only have read-only permission to this data.

**Comment by guglielmomoretti** [comment]
Ok so there are no viable workaround for this. Unless this bug is fixed, there are no way of reading out  specific key if they are in the format you described...

**Comment by busbaby** [comment]
I suppose one workaround would be to create a function to avoid using the range construct.
For example:
select MAPGET(map, "8d2eb047-d372-45e7-b74f-050fda046b33")

**Comment by luigidellaquila** [comment]
Hi all,

This issue is fixed in v 3.0 with the new SQL executor. I'm closing it

Thanks

Luigi
